pgmax/fg/groups.py

```diff
@@ -359,18 +359,47 @@ class FactorGroup:
         factor_type: Factor type shared by all the Factors in the FactorGroup.
         factor_configs: Optional array containing an explicit enumeration of all valid configurations
         log_potentials: Array of log potentials.
+
+    Attributes:
+        factor_sizes: Array of the different factor sizes.
+        variables_for_factors: Tuple concatenating the variables connected to each factor in the FactorGroup.
+            Each variable will appear once for each Factor it connects to.
+        factor_edges_num_states: Array concatenating the number of states for the variables connected to each Factor of
+            the FactorGroup. Each variable will appear once for each Factor it connects to.
+
+    Raises:
+        ValueError: if the FactorGroup does not contain a Factor
     """
 
     variable_group: Union[CompositeVariableGroup, VariableGroup]
     variable_names_for_factors: Sequence[List]
     factor_type: Type = field(init=False)
```

Comment:
`factor_type` should also be an attribute instead of an argument, since it's never explicitly specified at instiatiation